Adding Typescript snippets to context documentation (#1634)
google/adk-docs: docs/context/compaction.md

Reviewer: ```suggestion
The `LlmEventSummarizer` (Python/Java) or `LlmSummarizer` (TypeScript) class allows
you to specify a particular model for summarization.
```

        .build();
    ```

=== "TypeScript"

    ```typescript
    import {Gemini, LlmAgent, LlmSummarizer, TokenBasedContextCompactor} from '@google/adk';

    const agent = new LlmAgent({
      name: 'my-agent',

>       model: 'gemini-2.5-flash',

Reviewer: Let's use `gemini-flash-latest` here and all other occurrences throughout this PR for code samples. We made this change globally in #1594.

Author: Oh good to know- thanks for flagging! Fixed.

        .build();
    ```

=== "TypeScript"

    ```typescript
    import {Gemini, LlmAgent, LlmSummarizer, TokenBasedContextCompactor} from '@google/adk';

    // Define the AI model to be used for summarization:
    const summarizationLlm = new Gemini({model: 'gemini-2.5-flash'});

    // Create the summarizer with the custom model:
    const mySummarizer = new LlmSummarizer({llm: summarizationLlm});

    // Configure the agent with the custom summarizer and compaction settings:
    const agent = new LlmAgent({
      name: 'my-agent',

>       contextCompactors: [

Reviewer: This is missing a model field.

Author: fixed

        new TokenBasedContextCompactor({
          tokenThreshold: 1000,
          eventRetentionSize: 1,
          summarizer: mySummarizer,
        }),
      ],
    });
    ```

You can further refine the operation of the `SlidingWindowCompactor` (Python/Java) or `TokenBasedContextCompactor` (TypeScript)
by modifying its summarizer class `LlmEventSummarizer` (Python/Java) or `LlmSummarizer` (TypeScript),

> including changing the `prompt_template` (Python/Java) or `prompt` (TypeScript) setting of that class. For more details, see the

Reviewer: ```suggestion
You can further refine the compactor by modifying its summarizer. In Python and Java,
customize the `prompt_template` on `LlmEventSummarizer`. In TypeScript, customize
the `prompt` on `LlmSummarizer`. For more details, see the
```

Let's shorten and simplify this given all of the options in place now.

        .build();
    ```

=== "TypeScript"

    ```typescript
    import {Gemini, LlmAgent, LlmSummarizer, TokenBasedContextCompactor} from '@google/adk';

    // Define the AI model to be used for summarization:

>     const summarizationLlm = new Gemini({model: 'gemini-2.5-flash'});

Reviewer: ```suggestion
    const summarizationLlm = new Gemini({model: 'gemini-flash-latest'});
```
